packages/mollie-integration/src/mollie-payment-strategy.ts

```diff
@@ -168,59 +158,58 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
-            return this._processAdditionalAction(error);
+            await this._processAdditionalAction(error);
         }
     }
 
-    finalize(): Promise<InternalCheckoutSelectors> {
-        return Promise.resolve(this._store.getState());
+    finalize(): Promise<void> {
+        return Promise.resolve();
     }
 
-    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        if (this._unsubscribe) {
-            this._unsubscribe();
+    deinitialize(options?: PaymentRequestOptions): Promise<void> {
+        if (this.unsubscribe) {
+            this.unsubscribe();
         }
 
-        if (this._hostedForm) {
-            this._hostedForm.detach();
+        if (this.hostedForm) {
+            this.hostedForm.detach();
         }
 
-        if (options && options.methodId && options.gatewayId && !this._hostedForm) {
+        if (options && options.methodId && options.gatewayId && !this.hostedForm) {
             const element = document.getElementById(`${options.gatewayId}-${options.methodId}`);
 
             if (element) {
                 element.remove();
             }
         } else if (options && options.methodId && this.isCreditCard(options.methodId)) {
             if (
-                this._cardHolderElement &&
-                this._cardNumberElement &&
-                this._verificationCodeElement &&
-                this._expiryDateElement
+                this.cardHolderElement &&
+                this.cardNumberElement &&
+                this.verificationCodeElement &&
+                this.expiryDateElement
             ) {
-                this._cardHolderElement.unmount();
-                this._cardHolderElement = undefined;
+                this.cardHolderElement.unmount();
+                this.cardHolderElement = undefined;
 
-                this._cardNumberElement.unmount();
-                this._cardNumberElement = undefined;
+                this.cardNumberElement.unmount();
+                this.cardNumberElement = undefined;
 
-                this._verificationCodeElement.unmount();
-                this._verificationCodeElement = undefined;
+                this.verificationCodeElement.unmount();
+                this.verificationCodeElement = undefined;
 
-                this._expiryDateElement.unmount();
-                this._expiryDateElement = undefined;
+                this.expiryDateElement.unmount();
+                this.expiryDateElement = undefined;
             }
         }
 
-        this._mollieClient = undefined;
+        this.mollieClient = undefined;
 
-        return Promise.resolve(this._store.getState());
+        return Promise.resolve();
     }
 
-    protected async executeWithCC(
-        payment: OrderPaymentRequestBody,
-    ): Promise<InternalCheckoutSelectors> {
+    protected async executeWithCC(payment: OrderPaymentRequestBody): Promise<void> {
         const paymentData = payment.paymentData;
+        /* eslint-disable */
```

Comment:
maybe add some type guard or add correct type to the method argument?

Reply:
fixed